src/pubsub.c

```diff
@@ -255,13 +277,13 @@ int pubsubSubscribeChannel(client *c, robj *channel, pubsubtype type) {
         /* Add the client to the channel -> list of clients hash table */
         de = dictFind(*type.serverPubSubChannels, channel);
         if (de == NULL) {
-            clients = listCreate();
+            clients = dictCreate(&clientDictType);
             dictAdd(*type.serverPubSubChannels, channel, clients);
             incrRefCount(channel);
         } else {
             clients = dictGetVal(de);
         }
-        listAddNodeTail(clients,c);
+        serverAssert(dictAddRaw(clients, c, NULL) != NULL);
```

Comment:
i'd rather use a plain `dictAdd`, we don't need a reference to the new dictEntry or existing one.